File: src/rust/calculate_pi.rs
fn main(){
  println!("{:?}", calculate_pi(vec![10, 1000, 100000, 10000000]));
}
fn calculate_pi(terms: Vec<i32>) -> Vec<f64> {
  let mut denominator: f64;
  let mut operation: f64;
  let mut pi: Vec<f64> = Vec::<f64>::new();

  for i in 0..terms.len() {
    denominator = 1.0;
    operation = 1.0;
    pi.push(0.0);
    for _ in 0..terms[i] {
      let i: usize = i as usize;
      pi[i] += operation * (4.0 / denominator);
      denominator += 2.0;
      operation *= -1.0;
    }
  }
  pi
}

File: src/go/calculate_pi.go
package main

import "fmt"

func CalculatePi(terms []int) []float64 {
	var denominator float64
	var operation float64
	pi := make([]float64, len(terms))

	for i := 0; i < len(terms); i++ {
		denominator = 1.0
		operation = 1.0
		for cont := 0; cont < terms[i]; cont++ {
			pi[i] += operation * (4 / denominator)
			denominator += 2
			operation *= -1
		}
	}
	return pi
}

func main() {
	slice := []int{10, 1000, 100000, 10000000}
	fmt.Println("Slice:", slice)
	fmt.Println("Pi:", CalculatePi(slice))
}
